### ISSUE TITLE
Problem with RFID Service

### ISSUE THREAD

**Issue by derschelli** [MiczFlor/RPi-Jukebox-RFID#438]
Hello there,

I'trying to setup a Phoniebox, following the provided installation instructions.

I'm using the latest Stretch Version, installad on a Raspberry 3B (without+). RFID-Reader is the recommended device from Neuftech.

Everything went fine up until setting up the automatic start as described in https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/CONFIGURE-stretch

> To see if the reader process is running use the following command:
> 
> sudo systemctl status phoniebox-rfid-reader

results in the following output:

> pi@musikkiste:~ $ sudo systemctl status phoniebox-rfid-reader
>  phoniebox-rfid-reader.service - Phoniebox RFID-Reader Service
>    Loaded: loaded (/etc/systemd/system/phoniebox-rfid-reader.service; enabled; vendor preset: enabled)
>    Active: failed (Result: exit-code) since Sat 2019-01-12 13:54:28 CET; 37s ago
>   Process: 740 ExecStart=/home/pi/RPi-Jukebox-RFID/scripts/daemon_rfid_reader.py (code=exited, status=1/FAILURE)
>  Main PID: 740 (code=exited, status=1/FAILURE)
> 
> Jan 12 13:54:28 musikkiste systemd[1]: phoniebox-rfid-reader.service: Service hold-off time over, scheduling restart.
> Jan 12 13:54:28 musikkiste systemd[1]: Stopped Phoniebox RFID-Reader Service.
> Jan 12 13:54:28 musikkiste systemd[1]: phoniebox-rfid-reader.service: Start request repeated too quickly.
> Jan 12 13:54:28 musikkiste systemd[1]: Failed to start Phoniebox RFID-Reader Service.
> Jan 12 13:54:28 musikkiste systemd[1]: phoniebox-rfid-reader.service: Unit entered failed state.
> Jan 12 13:54:28 musikkiste systemd[1]: phoniebox-rfid-reader.service: Failed with result 'exit-code'.

The RFID-Reader is not working.
But if i start the rfid-reader service manually everything seems to work:

> pi@musikkiste:~ $ /home/pi/RPi-Jukebox-RFID/scripts/daemon_rfid_reader.py &
> [2] 2634

After that, the RFID-Reader is working fine, but i can't log in after every restart and start the rfid-deamon manually. Is there a solution to fix the service? This is the content of the file:

> pi@musikkiste:~ $ cat /etc/systemd/system/phoniebox-rfid-reader.service
> [Unit]
> Description=Phoniebox RFID-Reader Service
> After=network.target iptables.service firewalld.service
> 
> [Service]
> Restart=always
> WorkingDirectory=/home/pi/RPi-Jukebox-RFID
> ExecStart=/home/pi/RPi-Jukebox-RFID/scripts/daemon_rfid_reader.py
> 
> [Install]
> WantedBy=multi-user.target

Thank you :-)

**Comment by derschelli** [comment]
I found the solution: i had to add User=pi and Group=pi, similar to the other service files:

> [Service]
> User=pi
> Group=pi
> Restart=always
> WorkingDirectory=/home/pi/RPi-Jukebox-RFID
> ExecStart=/home/pi/RPi-Jukebox-RFID/scripts/daemon_rfid_reader.py

**Comment by MiczFlor** [comment]
Hi @derschelli 
I fixed this for all services. Strange that it didn't work for you, but now it's on the safe side.

**Comment by peterwilc** [comment]
Hello, I was getting this exact error message using the buster install on a raspberry pi 4.
I fixed it by converting the daemon_rfid_reader.py file to be python3 compatible. When I would try to run it in python2, it gave me an error that it could not find evdev even though evdev was clearly installed. After I converted the file to be python3 compatible, I restarted my RPi and when I ran system status on phoniebox-rfid-reader, I found that it worked and is active. I am a novice to coding so I don't have the insight to know what this means.
